src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -3972,8 +3973,10 @@ NEVER_INLINE void ByteCodeInterpreter::setObjectOpcodeSlowCase(ExecutionState& s
     if (willBeObject.isPrimitive()) {
         obj->preventExtensions(state);
     }
-
     bool result = obj->setIndexedProperty(state, property, registerFile[code->m_loadRegisterIndex]);
+    if (obj->isTypedArrayObject() && obj->asTypedArrayObject()->buffer()->isDetachedBuffer()) {
+        result = true;
+    }
```

Comment:
IMO this part can be removed because you already fixed `setIndexedProperty` of TypedArray according to the new standard.